### ISSUE TITLE
Make xcm pint transfers to statemint safe

### ISSUE THREAD

**Issue by mattsse** [ChainSafe/PINT#225]
## Issue summary

> Due to the way Xcm::Transact currently works, there is no way to get the response of a dispatched call of the target chain, so this currently assumes that the call succeeds. This leads to significant caveats, for example pallet_assets calls will fail if the beneficiary account does not exist yet (no existential deposit), so in order to receive minted statemint PINT (SPINT?) the account that calls transfer_to_statemint must have already an account on statemint. This means essentially: if the call fails on statemint, the funds are locked in the Statemintcustodian account on PINT.

https://github.com/ChainSafe/PINT/pull/222#issue-694275385

Also since the PINT sovereign account gets charged for this transact we need to prevent it from running dry, either by
- paying with SPINT, that we mint, requires that statemint accepts SPINT as payment
- combination of Xcm::Deposits and Xcm::Transact:
   1. charge the user in relay chain currency accordingly. Paying for XCM with relay chain currency that was contributed to the index will technically reduce the NAV of the relay chain currency, so, to keep the PINT NAV stable, the corresponding PINT NAV should probably be burned off the user's balance.
   2. reserve transfer -> PINT -> relay -> statemint to fund the PINT sovereign account on statemint
   3. transact the `assets` call that is now paid for 

## Other information and links

- Ref #173
- Ref #218
- Blocked by https://github.com/paritytech/xcm-format

- Ref #234

**Comment by mattsse** [comment]
closing in favor of #234
